packages/test/test-end-to-end-tests/src/test/intervalCollectionEndToEndTests.spec.ts

```diff
@@ -0,0 +1,141 @@
+			0,
+			sharedString.getLength() - 1,
+		);
+		assert.deepEqual(actual, overlapping, "Interval search returned inconsistent results");
+	}
+	assert.strictEqual(
+		actual.length,
+		expected.length,
+		`findOverlappingIntervals() must return the expected number of intervals`,
+	);
+
+	const actualPos = actual.map((interval) => {
+		assert(interval);
+		const start = sharedString.localReferencePositionToPosition(interval.start);
+		const end = sharedString.localReferencePositionToPosition(interval.end);
+		return { start, end };
+	});
+	assert.deepEqual(actualPos, expected, "intervals are not as expected");
+};
+
+describeNoCompat("IntervalCollection with stashed ops", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	let sharedString1: SharedString;
+	let sharedString2: SharedString;
+	let dataObject1: ITestFluidObject;
+	let dataObject2: ITestFluidObject;
+	let collection1: IIntervalCollection<SequenceInterval>;
+	let collection2: IIntervalCollection<SequenceInterval>;
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider();
+		const container1 = await provider.makeTestContainer(testContainerConfig);
+		dataObject1 = await getContainerEntryPointBackCompat<ITestFluidObject>(container1);
+		sharedString1 = await dataObject1.getSharedObject<SharedString>(stringId);
+		sharedString1.insertText(0, "hello world");
+		collection1 = sharedString1.getIntervalCollection(collectionId);
+
+		const container2 = await provider.loadTestContainer(testContainerConfig);
+		dataObject2 = await getContainerEntryPointBackCompat<ITestFluidObject>(container2);
+		sharedString2 = await dataObject2.getSharedObject<SharedString>(stringId);
+	});
+
+	it("doesn't resend successful op", async () => {
+		// add an interval
+		const id = collection1.add({ start: 4, end: 7 }).getIntervalId();
+
+		// pending ops stuff from e2e tests - make a new container, pause op processing,
+		// make a change, close the container, then resume op processing and reload container
+		const container: IContainerExperimental =
+			await provider.loadTestContainer(testContainerConfig);
+		await waitForContainerConnection(container);
+		const dataStore = (await container.getEntryPoint()) as ITestFluidObject;
+
+		[...Array(30).keys()].map((i) =>
+			dataStore.root.set(`make sure csn is > 1 so it doesn't hide bugs ${i}`, i),
+		);
+
+		await provider.ensureSynchronized();
+		await provider.opProcessingController.pauseProcessing(container);
+		assert(dataStore.runtime.deltaManager.outbound.paused);
+
+		// the "callback" portion of the original e2e test
+		const sharedString = await dataStore.getSharedObject<SharedString>(stringId);
+		const collection = sharedString.getIntervalCollection(collectionId);
+		collection.change(id, 3, 8);
+
+		const pendingState: string | undefined = await container.closeAndGetPendingLocalState?.();
+		provider.opProcessingController.resumeProcessing();
+		assert.ok(pendingState);
+
+		collection1.change(id, 2, 9);
```

Comment:
Not sure if there's something specific to do to reload the container here...the test passes, but I'm not positive that it recreates the scenario of the other e2e test exactly.

Reply:
it would be good to ensure that container 1 never sends the op we expect to be stashed. you could load the container without stashed ops, and ensure the interval is where it should be. then load a container with stashed ops to which should see the change. you can also keep the second container open, and ensure it also sees the op that was stashed and applied.